test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs

```diff
@@ -68,5 +68,31 @@ public void ExportResultIsSuccess(OtlpExportProtocol protocol, string endpoint)
             Assert.Single(delegatingExporter.ExportResults);
             Assert.Equal(ExportResult.Success, delegatingExporter.ExportResults[0]);
         }
+
+        [Trait("CategoryName", "CollectorIntegrationTests")]
+        [SkipUnlessEnvVarFoundFact(CollectorHostnameEnvVarName)]
+        public void ConstructingGrpcExporterFailsWhenHttp2UnencryptedSupportIsDisabledForNetcoreapp31()
+        {
+            // Adding the OtlpExporter creates a GrpcChannel.
+            // This switch must be set before creating a GrpcChannel/HttpClient when calling an insecure gRPC service.
+            // We want to fail fast so we are disabling it
+            // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
+            AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", false);
+
+            var exporterOptions = new OtlpExporterOptions
+            {
+                Endpoint = new Uri($"http://{CollectorHostname}:4317"),
+            };
+
+            if (Environment.Version.Major == 3)
+            {
+                Assert.Throws<InvalidOperationException>(() => new OtlpTraceExporter(exporterOptions));
+            }
+            else
+            {
+                var exporter = new OtlpTraceExporter(exporterOptions);
+                Assert.NotNull(exporter);
+            }
```

Comment:
how about using `Record.Exception` and have `Assert.IsType<T>` and `Assert.Null` depending on `Environment.Version.Major == 3`  ?

Reply:
Yes